src/Microsoft.Azure.WebJobs.Host/Bindings/BindingContext.cs

```diff
@@ -83,6 +86,18 @@ public TraceWriter Trace
         }
 
         /// <summary>
+        /// Gets the set of property values for the current bind operation.
+        /// </summary>
+        /// <remarks>
+        /// Bindings can add arbitrary values to this collection as needed to
+        /// flow state around.
+        /// </remarks>
+        public IDictionary<string, object> Properties
```

Comment:
This is the planned way for binding specific custom resolvers to communicate state back to the binding pipeline.